monix-reactive/shared/src/main/scala/monix/reactive/Consumer.scala

```diff
@@ -125,6 +128,18 @@ trait Consumer[-In, +R] extends ((Observable[In]) => Task[R])
     new MapAsyncConsumer[In,R,R2](self, f)
 }
 
+trait PartitionedConsumer[A] {
```

Comment:
It's true that `Int` is obscure, but you could add a type alias for it. Something like `type PartitionIndex = Int`. That should be enough for documentation purposes.